MSAL/test/unit/native_auth/public/delegate/sign_up/SignUpPasswordStartDelegateDispatcherTests.swift

```diff
@@ -0,0 +1,150 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+@testable import MSAL
+import XCTest
+
+final class SignUpPasswordStartDelegateDispatcherTests: XCTestCase {
+
+    private var telemetryExp: XCTestExpectation!
+    private var delegateExp: XCTestExpectation!
+    private var sut: SignUpPasswordStartDelegateDispatcher!
+    private let controllerFactoryMock = MSALNativeAuthControllerFactoryMock()
+    private let correlationId = UUID()
+
+    override func setUp() {
+        super.setUp()
+        telemetryExp = expectation(description: "delegateDispatcher telemetry exp")
+        delegateExp = expectation(description: "delegateDispatcher delegate exp")
+    }
+
+    func test_dispatchSignUpCodeRequired_whenDelegateMethodsAreImplemented() async {
+        let delegate = SignUpPasswordStartDelegateSpy(expectation: delegateExp)
+
+        sut = .init(delegate: delegate, telemetryUpdate: { result in
+            guard case .success = result else {
+                return XCTFail("wrong result")
+            }
+            self.telemetryExp.fulfill()
+        })
+
+        let expectedState = SignUpCodeRequiredState(controller: controllerFactoryMock.signUpController, username: "", flowToken: "flowToken", correlationId: correlationId)
+        let expectedSentTo = "user@contoso.com"
+        let expectedChannelTargetType = MSALNativeAuthChannelType.email
+        let expectedCodeLength = 4
+
+        await sut.dispatchSignUpCodeRequired(
```

Comment:
Is this call correctly being made when the tests are for PasswordStart?
Also the unit test name `test_dispatchSignUpCodeRequired_whenDelegateMethodsAreImplemented` is it correct?

Reply:
yes, the call is correct, but I think I see where the confusion comes from. I've renamed the method in the `SignUpPasswordStartDelegateDispatcher` to `dispatchSignUpPasswordCodeRequired` to make it clearer.

Same for the test names.